### PR TITLE
Initial Apollo commit

### PR DESCRIPTION
**Pull request by miriambudayr**
Following the Hasura documentation for auth0-spa-js rules, I couldn't get the authorization to work with the recommended access token. See the documentation [here](https://hasura.io/docs/1.0/graphql/manual/guides/integrations/auth0-jwt.html#configure-auth0-rules-callback-urls). We are supposed to use `getTokenSilently` from the Auth0 client.

I used the `idToken` instead which works, though I would like to figure out how to use the access token instead. 

Thanks for looking this over!

### CODE REVIEW

File: site/src/components/Auth0Provider.js [hunk 1]
```diff
@@ -59,6 +60,8 @@ export const Auth0Provider = ({
 
       if (localIsAuthenticated) {
         setUser(await auth0FromHook.getUser())
+        const idToken = await auth0FromHook.getIdTokenClaims()
+        localStorage.setItem("idToken", idToken.__raw)
```
Comment (bloudermilk) [inline site/src/components/Auth0Provider.js:64]: Given `AuthorizedApolloProvider` is wrapped in the `Auth0Provider` I think we can avoid putting this in localstorage.

File: site/src/components/AuthorizedApolloProvider.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from "react"
+import { ApolloClient, ApolloLink, InMemoryCache, HttpLink } from "apollo-boost"
+import { ApolloProvider } from "react-apollo"
+import { setContext } from "apollo-link-context"
+
+export const AuthorizedApolloProvider = ({ children }) => {
```
Comment (bloudermilk) [inline site/src/components/AuthorizedApolloProvider.js:6]: We should be able to use the context here. See `LoginPage` for an example.

**Comment by bloudermilk** [comment]
Looking further down in that documentation you linked, it looks like we'll also need to make a change to our `Auth0Provider` to include the `audience` [as shown here](https://hasura.io/docs/1.0/graphql/manual/guides/integrations/auth0-jwt.html#create-an-auth0-api)

**Comment by miriambudayr** [comment]
@bloudermilk 
Yeah, I had used the setup with `audience`/`accessToken` but it didn't authorize when I ran a test query. I will try again and see if it works in the production build.

**Pull request merged**